pom.xml

```diff
@@ -538,11 +538,24 @@
             <version>1.9.13</version>
         </dependency>
 
-        <!-- hadoop compilation dependency for CodecPool -->
+        <!-- hadoop compilation dependencies for CodecPool and AvroGenericRecordReader -->
         <dependency>
             <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-core</artifactId>
-            <version>0.20.2</version>
+            <artifactId>hadoop-common</artifactId>
+            <version>3.1.0</version>
+            <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-mapreduce-client-core</artifactId>
+            <version>3.1.0</version>
```

Comment:
Add a `dep.hadoop.version` property for this since we want the versions to be the same